velox/dwio/common/tests/utils/FilterGenerator.cpp

```diff
@@ -141,46 +142,57 @@ std::unique_ptr<Filter> ColumnStats<float>::makeRangeFilter(
       upper,
       upperUnbounded,
       false,
-      selectPct > 25);
+      filterSpec.selectPct > 25);
 }
 
 template <>
 std::unique_ptr<Filter> ColumnStats<double>::makeRangeFilter(
-    float startPct,
-    float selectPct) {
+    FilterSpec filterSpec) {
   if (values_.empty()) {
-    return std::make_unique<velox::common::IsNull>();
+    return std::make_unique<velox::common::DoubleRange>(
```

Comment:
Was there any reason you changed the filter for empty values for this double? For other types there are sometimes checking on allowNulls_, sometimes not. If there are no specific reasons, let's revert the changes. In my understanding, the allowNulls_ doesn't matter for empty values.

Reply:
I see your point. Before introducing this PR, the double filter returns true all the nulls. And also, if all the elements in the vector are nulls, `values_.empty()` returns true. In the current PR, we need it to return false on all the nulls, so we need it here.